app/controllers/application_controller/tags.rb

```diff
@@ -86,6 +86,15 @@ def tagging_tags_set_form_vars
     @edit[:current] = copy_hash(@edit[:new])
   end
 
+  def previous_page_url
```

Comment:
This is a generic name for a method that could get mixed into many forms.  I think this method should live right next to previous_breadcrumb_url, wherever that's defined, since it's closely related to it.